lib/data/model.ts

```diff
@@ -124,13 +124,54 @@ export default class Model extends DenaliObject {
     assert(container instanceof Container, `You must supply a container to lookup this model's adapter instead`);
     let adapter = container.lookup(`orm-adapter:${ this.type }`, { loose: true });
     if (!adapter) {
+      // Specific model adapter not found, try application adapter
       adapter = container.lookup('orm-adapter:application', { loose: true });
     }
     assert(adapter, `No orm-adapter found for "${ this.type }", and no fallback "application" orm-adapter found either. Available adapters: ${ container.availableForType('orm-adapter') }`);
     return adapter;
   }
 
   /**
+   * Call the supplied callback function for each attribute on this model, passing in the attribute
+   * name and attribute instance.
+   */
+  static mapAttributes<T>(container: Container, fn: (descriptor: Attribute, attributeName: string) => T): T[] {
```

Comment:
So the reason for making them static is twofold:

1) ORM adapters need to be able to loop through attributes and relationships on the model class in order to get set up correctly. This is the only way of doing it. The orm-adapter could create an instance of each model, but even then it wouldn't have access to the actual attribute/relationship descriptor unless it re-implemented the static behavior
2) An instance can call the static method, then get the instance value using the key of the attribute/relationship, but an orm can't infer the attribute descriptor from the instance value.